spec/meilisearch/models/task_spec.rb

```diff
@@ -0,0 +1,485 @@
+# frozen_string_literal: true
+
+describe MeiliSearch::Models::Task do
+  let(:new_index_uid) { random_uid }
+  let(:task_hash) { client.http_post '/indexes', { 'uid' => new_index_uid } }
+  let(:endpoint) { MeiliSearch::Task.new(URL, MASTER_KEY, client.options) }
+
+  subject { described_class.new task_hash, endpoint }
+
+  let(:enqueued_endpoint) { instance_double(MeiliSearch::Task, task: task_hash) }
+  let(:enqueued_task) { described_class.new task_hash, enqueued_endpoint }
+
+  let(:processing_endpoint) { instance_double(MeiliSearch::Task, task: task_hash.update('status' => 'processing')) }
+  let(:processing_task) { described_class.new task_hash, processing_endpoint }
+
+  let(:logger) { instance_double(Logger, warn: nil) }
+  before { MeiliSearch::Utils.logger = logger }
+  after { MeiliSearch::Utils.logger = nil }
+
+  describe '.initialize' do
+    it 'requires a uid in the task hash' do
+      task_hash.delete 'taskUid'
+
+      expect { subject }.to raise_error(ArgumentError)
+    end
+
+    it 'requires a type in the task hash' do
+      task_hash.delete 'type'
+
+      expect { subject }.to raise_error(ArgumentError)
+    end
+
+    it 'requires a status in the task hash' do
+      task_hash.delete 'status'
+
+      expect { subject }.to raise_error(ArgumentError)
+    end
+
+    it 'sets "taskUid" key when given a "uid"' do
+      expect(subject).to have_key('uid')
+    end
+
+    it 'sets "uid" key when given a "taskUid"' do
+      task_hash['uid'] = task_hash.delete 'taskUid'
+
+      expect(subject).to have_key('taskUid')
+    end
+  end
+
+  describe 'forwarding' do
+    it 'allows accessing values in the internal task hash' do
+      subject
+
+      task_hash.each do |key, value|
+        expect(subject[key]).to eq(value)
+      end
+    end
+  end
+
+  describe '#enqueued?' do
+    context 'when the task is processing' do
+      before { task_hash['status'] = 'processing' }
+
+      it { is_expected.not_to be_enqueued }
+
+      it 'does not refresh the task' do
+        allow(subject).to receive(:refresh)
+        subject.enqueued?
+        expect(subject).not_to have_received(:refresh)
```

Comment:
When I mention the xUnit I mean this:

```suggestion
        allow(subject).to receive(:refresh)
        
        subject.enqueued?
        
        expect(subject).not_to have_received(:refresh)
```